### ISSUE TITLE
arxiv: 'getUpdated' now gets the wrong updated field (the feed, not article)

### ISSUE THREAD

**Issue by gwern** [toschoo/Haskell-Libs#1]
Investigating incorrect metadata on some of my Arxiv links leading to incorrect anchors (why was [Hestness et al 2017](https://arxiv.org/abs/1712.00409#baidu "Deep Learning Scaling is Predictable, Empirically") turning into an ID anchor of `#hestness-et-al-2020`, breaking my links to `#hestness-et-al-2017`, which is where I expected it to be as it hasn't been updated since?), it seems that Arxiv now has two fields in its API exports named `updated`, and I assume when you wrote the `arxiv` library it only had one or something:

~~~{.XML}
$ curl --location --silent "https://export.arxiv.org/api/query?search_query=id:1712.00409&start=0&max_results=1" --user-agent "gwern+arxivscraping@gwern.net"
<?xml version="1.0" encoding="UTF-8"?>
<feed xmlns="http://www.w3.org/2005/Atom">
  <link href="http://arxiv.org/api/query?search_query%3Did%3A1712.00409%26id_list%3D%26start%3D0%26max_results%3D1" rel="self" type="application/atom+xml"/>
  <title type="html">ArXiv Query: search_query=id:1712.00409&amp;id_list=&amp;start=0&amp;max_results=1</title>
  <id>http://arxiv.org/api/w+IZ+y+HsvaY7cE7nfbIUd/Sml8</id>
  <updated>2020-11-04T00:00:00-05:00</updated>
  <opensearch:totalResults xmlns:opensearch="http://a9.com/-/spec/opensearch/1.1/">3</opensearch:totalResults>
  <opensearch:startIndex xmlns:opensearch="http://a9.com/-/spec/opensearch/1.1/">0</opensearch:startIndex>
  <opensearch:itemsPerPage xmlns:opensearch="http://a9.com/-/spec/opensearch/1.1/">1</opensearch:itemsPerPage>
  <entry>
    <id>http://arxiv.org/abs/1712.00409v1</id>
    <updated>2017-12-01T17:13:14Z</updated>
    <published>2017-12-01T17:13:14Z</published>
    <title>Deep Learning Scaling is Predictable, Empirically</title>
    <summary>  Deep learning (DL) creates impactful advances following a virtuous recipe:
model architecture search, creating large training data sets, and scaling
computation. It is widely believed that growing training sets and models should
improve accuracy and result in better products. As DL application domains grow,
we would like a deeper understanding of the relationships between training set
size, computational scale, and model accuracy improvements to advance the
state-of-the-art.
  This paper presents a large scale empirical characterization of
generalization error and model size growth as training sets grow. We introduce
a methodology for this measurement and test four machine learning domains:
machine translation, language modeling, image processing, and speech
recognition. Our empirical results show power-law generalization error scaling
across a breadth of factors, resulting in power-law exponents---the "steepness"
of the learning curve---yet to be explained by theoretical work. Further, model
improvements only shift the error but do not appear to affect the power-law
exponent. We also show that model size scales sublinearly with data size. These
scaling relationships have significant implications on deep learning research,
practice, and systems. They can assist model debugging, setting accuracy
targets, and decisions about data set growth. They can also guide computing
system design and underscore the importance of continued computational scaling.
</summary>
    <author>
      <name>Joel Hestness</name>
    </author>
    <author>
      <name>Sharan Narang</name>
    </author>
    <author>
      <name>Newsha Ardalani</name>
    </author>
    <author>
      <name>Gregory Diamos</name>
    </author>
    <author>
      <name>Heewoo Jun</name>
    </author>
    <author>
      <name>Hassan Kianinejad</name>
    </author>
    <author>
      <name>Md. Mostofa Ali Patwary</name>
    </author>
    <author>
      <name>Yang Yang</name>
    </author>
    <author>
      <name>Yanqi Zhou</name>
    </author>
    <arxiv:comment xmlns:arxiv="http://arxiv.org/schemas/atom">19 pages, 11 figures</arxiv:comment>
    <link href="http://arxiv.org/abs/1712.00409v1" rel="alternate" type="text/html"/>
    <link title="pdf" href="http://arxiv.org/pdf/1712.00409v1" rel="related" type="application/pdf"/>
    <arxiv:primary_category xmlns:arxiv="http://arxiv.org/schemas/atom" term="cs.LG" scheme="http://arxiv.org/schemas/atom"/>
    <category term="cs.LG" scheme="http://arxiv.org/schemas/atom"/>
    <category term="stat.ML" scheme="http://arxiv.org/schemas/atom"/>
  </entry>
</feed>
~~~

ie.

~~~{.XML}
  <updated>2020-11-04T00:00:00-05:00</updated>
...
<entry>
...    <updated>2017-12-01T17:13:14Z</updated>
~~~

So [when I run](https://www.gwern.net/static/build/LinkMetadata.hs) curl and then import the XML into arxiv and run `getUpdated`, it returns the first, which is the wrong one:

~~~{.Haskell}
Just ("https://arxiv.org/abs/1712.00409#baidu",("Deep Learning Scaling is Predictable, Empirically","Joel Hestness, Sharan Narang, Newsha Ardalani, Gregory Diamos, Heewoo Jun, Hassan Kianinejad, Md. Mostofa Ali Patwary, Yang Yang, Yanqi Zhou","2020-11-04","","<p>Deep learning (DL) creates impactful advances following a virtuous recipe: model architecture search, creating large training data sets, and scaling computation. It is widely believed that growing training sets and models should improve accuracy and result in better products. As DL application domains grow, we would like a deeper understanding of the relationships between training set size, computational scale, and model accuracy improvements to advance the state-of-the-art.</p><p>This paper presents a large scale empirical characterization of generalization error and model size growth as training sets grow. We introduce a methodology for this measurement and test four machine learning domains: machine translation, language modeling, image processing, and speech recognition. Our empirical results show power-law generalization error scaling across a breadth of factors, resulting in power-law exponents\8212the \"steepness\" of the learning curve\8212yet to be explained by theoretical work. Further, model improvements only shift the error but do not appear to affect the power-law exponent. We also show that model size scales sublinearly with data size. These scaling relationships have significant implications on deep learning research, practice, and systems. They can assist model debugging, setting accuracy targets, and decisions about data set growth. They can also guide computing system design and underscore the importance of continued computational scaling.</p>"))
~~~

 I'll switch to `getPublished` which is probably more accurate for my purpose anyway, but `getUpdated` seems like it needs to get updated. :)

**Comment by toschoo** [comment]
Hi gwern,
Thanks for the feedback.
If I understand correctly, you want <entry><updated></entry>, but you get the top-level <updated>. Is that correct?
To get the contents within the entries, you need to enter the entry before you call getUpdated, for instance by using one of forEach, forEachM, forEachM_
Hope that helps...

**Comment by gwern** [comment]
But why would any user expect to need to do that? It works and runs without any issue. It's just it returns something completely useless (someone issuing a request knows exactly what it was created, because they made it!), without any hint that it's the wrong thing. And note that the documentation very clearly promises the user that it does the right expected thing and does *not* give you the 'updated' date of the *search query* but the *article* (emphasis added):

> Gets the contents of the "update" field in this entry, i.e. the date when the **article** was last updated.

**Comment by toschoo** [comment]
I would emphasise differently:

> Gets the contents of the "update" field in **this** entry, i.e. the date when the article was last updated.

**Comment by gwern** [comment]
A search query is not an 'article', no matter what letters you choose to emphasize. The code and documentation are wrong. So this is a bug. Period.

If you are not maintaining this code, don't play games with people who are going out of their way to report bugs in good faith; simply say so, mark it WONTFIX, and archive the repo as abandoned so I know I will have to write my own Arxiv processing code (because this library will continue to bitrot).

**Comment by toschoo** [comment]
Perhaps you could actually read the documentation.
